meson.build

```diff
@@ -2196,7 +2196,7 @@ test_additional_kwargs = {
 }
 
 fuzz_template = executable_template + {
-        'build_by_default' : fuzzer_build,
+        'build_by_default' : want_fuzz_tests,
```

Comment:
Looking at https://github.com/systemd/systemd/actions/runs/15088857024/job/42414867661 (which was run against the main branch) it seems the binaries like `fuzz-json` were built and run so it seems it doesn't change anything in the sense that `build_by_default` doesn't seem to affect the binaries.

The "build_by_default" tests are handled in https://github.com/systemd/systemd/blob/741a184a31127305fb4363833ca9d97a1057fc68/meson.build#L2505-L2511

but as far as can see the fuzz targets are handled separately.

Reply:
The way `build_by_default` is handled in different places should probably be addressed separately so I'll go ahead and resolve this conversation.

I'm not sure `want_fuzz_tests` should take precedence over `fuzzer_build` because the flags can be set/unset separately but it's a separate issue too probably.